src/packages/engine-core/src/NAPIEngine.ts

```diff
@@ -0,0 +1,547 @@
+    queries: string[],
+    transaction = false,
+    numTry = 1,
+  ): Promise<any> {
+    await this.start()
+    const variables = {}
+    const body = {
+      batch: queries.map((query) => ({ query, variables })),
+      transaction,
+    }
+    const result = await this.engine!.query(body)
+    const data = JSON.parse(result)
+
+    if (data.errors) {
+      if (data.errors.length === 1) {
+        throw this.graphQLToJSError(data.errors[0])
+      }
+      // this case should not happen, as the query engine only returns one error
+      throw new PrismaClientUnknownRequestError(
+        JSON.stringify(data.errors),
+        this.config.clientVersion!,
+      )
+    }
+    try {
+      const { batchResult, errors } = data
+      if (Array.isArray(batchResult)) {
+        return batchResult.map((result) => {
+          if (result.errors) {
+            return this.graphQLToJSError(result.errors[0])
+          }
+          return {
+            data: result,
+          }
+        })
+      } else {
+        if (errors && errors.length === 1) {
+          throw new Error(errors[0].error)
+        }
+        throw new Error(JSON.stringify(data))
+      }
+    } catch (e) {
+      // retry
+      if (numTry <= MAX_REQUEST_RETRIES) {
+        return this.requestBatch(queries, transaction, numTry + 1)
+      }
+
+      throw e
+    }
+  }
+
+  private async resolveEnginePath(): Promise<{
+    enginePath: string
+    searchedLocations: string[]
+  }> {
+    const searchedLocations: string[] = []
+    let enginePath
+    if (this.libQueryEnginePath) {
+      return { enginePath: this.libQueryEnginePath, searchedLocations }
+    }
+
+    this.platform = this.platform ?? (await getPlatform())
+
+    if (__filename.includes('NAPIEngine')) {
+      enginePath = path.join(getEnginesPath(), getNapiName(this.platform, 'fs'))
+      return { enginePath, searchedLocations }
+    }
+    const searchLocations: string[] = [
+      eval(`require('path').join(__dirname, '../../../.prisma/client')`), // Dot Prisma Path
+      this.config.generator?.output ?? eval('__dirname'), // Custom Generator Path
+      path.join(eval('__dirname'), '..'), // parentDirName
+      path.dirname(this.config.datamodelPath), // Datamodel Dir
+      this.config.cwd, //cwdPath
+    ]
+
+    if (this.config.dirname) {
+      searchLocations.push(this.config.dirname)
+    }
+
+    for (const location of searchLocations) {
+      searchedLocations.push(location)
+      debug(`Search for Query Engine in ${location}`)
+      enginePath = path.join(location, getNapiName(this.platform, 'fs'))
+      if (fs.existsSync(enginePath)) {
+        return { enginePath, searchedLocations }
+      }
+    }
+    enginePath = path.join(__dirname, getNapiName(this.platform, 'fs'))
+
+    return { enginePath: enginePath ?? '', searchedLocations }
+  }
+  private async getLibQueryEnginePath(): Promise<string> {
+    const libPath = process.env.LIB_QUERY_ENGINE_PATH ?? this.config.prismaPath
```

Comment:
Could we rename to something like 
PRISMA_LIB_QUERY_ENGINE_BINARY / PATH ? So it's more aligned with current "PRISMA_QUERY_ENGINE_BINARY"

Reply:
Sure

Reply:
Changed to `PRISMA_QUERY_ENGINE_LIBRARY` as it is a library and not a BINARY